src/x/instrument/options.go

```diff
@@ -63,6 +63,14 @@ func NewOptions() Options {
 	}
 }
 
+// NewTaggedOptions returns a new Options with the metric and logging client
+// tagged with the provided key value.
+func NewTaggedOptions(iOpts Options, k, v string) Options {
+	scope := iOpts.MetricsScope().Tagged(map[string]string{k: v})
+	logger := iOpts.Logger().With(zap.String(k, v))
```

Comment:
Hm, I would probably make the logger call separate so that you aren't forced to always have the logger add a ton of tags.

i.e. `NewMetricsTaggedOptions(....)` and `NewLogTaggedOptions(...)`

In fact what would probably be best is use the option param pattern here, API with something like this:
```go

iopts := instrument.WithOptions(iopts, 
  instrument.MetricsSubScopeOption("componentName"),
  instrument.MetricsTagOption("foo", "bar"),
  instrument.MetricsTagOption("baz", "qux"),
  instrument.LogTagOption("baz", "qux"))
```